### PR TITLE
[mecab] Add new port

### PR DESCRIPTION
**Pull request by xuchaoUCAS**

### CODE REVIEW

**Comment by xuchaoUCAS** [comment]
The open-source Mecab project does not support uwp/arm/arm64 platforms

File: ports/mecab/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+#   CURRENT_PACKAGES_DIR      = ${VCPKG_ROOT_DIR}\packages\${PORT}_${TARGET_TRIPLET}
+#   CURRENT_PORT_DIR          = ${VCPKG_ROOT_DIR}\ports\${PORT}
+#   PORT                      = current port name (zlib, etc)
+#   TARGET_TRIPLET            = current triplet (x86-windows, x64-windows-static, etc)
+#   VCPKG_CRT_LINKAGE         = C runtime linkage type (static, dynamic)
+#   VCPKG_LIBRARY_LINKAGE     = target library linkage type (static, dynamic)
+#   VCPKG_ROOT_DIR            = <C:\path\to\current\vcpkg>
+#   VCPKG_TARGET_ARCHITECTURE = target architecture (x64, x86, arm)
+#
+
```
Comment (ehsan-mohammadi) [inline ports/mecab/portfile.cmake]: I'm not sure, but can you add some codes like this to check `UWP`, `ARM` and `ARM64`?
```suggestion
if (UWP OR ARM OR ARM64)
    message(FATAL_ERROR "mecab does not support on this platform")
```

Comment (dan-shaw) [reply ports/mecab/portfile.cmake]: `VCPKG_TARGET_ARCHITECTURE` and `VCPKG_TARGET_IS_UWP` would work.

File: ports/mecab/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+include(vcpkg_common_functions)
+
+set(MECAB_VERSION 0.996)
+vcpkg_download_distfile(ARCHIVE
+    URLS "https://github.com/taku910/mecab/archive/master.zip"
```
Comment (dan-shaw) [inline ports/mecab/portfile.cmake]: You could use `vcpkg_from_github`.

File: ports/mecab/portfile.cmake (hunk 1 above)
Comment (xuchaoUCAS) [reply ports/mecab/portfile.cmake]: added related code

File: ports/mecab/portfile.cmake (hunk 2 above)
Comment (xuchaoUCAS) [reply ports/mecab/portfile.cmake]: replaced vcpkg_download_distfile with vcpkg_from_github

File: ports/mecab/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+#   CURRENT_PORT_DIR          = ${VCPKG_ROOT_DIR}\ports\${PORT}
+#   PORT                      = current port name (zlib, etc)
+#   TARGET_TRIPLET            = current triplet (x86-windows, x64-windows-static, etc)
+#   VCPKG_CRT_LINKAGE         = C runtime linkage type (static, dynamic)
+#   VCPKG_LIBRARY_LINKAGE     = target library linkage type (static, dynamic)
+#   VCPKG_ROOT_DIR            = <C:\path\to\current\vcpkg>
+#   VCPKG_TARGET_ARCHITECTURE = target architecture (x64, x86, arm)
+#
+
+if (UWP OR ARM OR ARM64)
```
Comment (ehsan-mohammadi) [inline ports/mecab/portfile.cmake]: Change this line to:
```suggestion
if (TARGET_TRIPLET STREQUAL "x64-uwp" OR TARGET_TRIPLET STREQUAL "arm64-windows" OR TARGET_TRIPLET STREQUAL "arm-uwp")
```

Comment (xuchaoUCAS) [reply ports/mecab/portfile.cmake]: updated.

**Comment by xuchaoUCAS** [comment]
Hi team, are there anything I need to do before merge into master?

File: ports/mecab/portfile.cmake [hunk 4]
```diff
@@ -10,7 +10,7 @@
 #   VCPKG_TARGET_ARCHITECTURE = target architecture (x64, x86, arm)
 #
 
-if (UWP OR ARM OR ARM64)
+if (TARGET_TRIPLET STREQUAL "x64-uwp" OR TARGET_TRIPLET STREQUAL "arm64-windows" OR TARGET_TRIPLET STREQUAL "arm-uwp")
```
Comment (ehsan-mohammadi) [inline ports/mecab/portfile.cmake]: I'm so sorry. I did a foolish mistake! Try to change this line to:

```suggestion
if (VCPKG_TARGET_TRIPLET STREQUAL "x64-uwp" OR VCPKG_TARGET_TRIPLET STREQUAL "arm64-windows" OR VCPKG_TARGET_TRIPLET STREQUAL "arm-uwp")
```

Sorry for my bad mistake. It should pass all the checks :)

**Review by ehsan-mohammadi — CHANGES_REQUESTED** [review]

File: ports/mecab/portfile.cmake (hunk 4 above)
Comment (xuchaoUCAS) [reply ports/mecab/portfile.cmake]: changed this line, but still cannot pass all the checks :(

**Comment by Rastaban** [comment]
It looks like the remaining build failures are by design, so they are okay.

**Pull request merged**